File: Stack/stack.py
class Stack:
    def __init__(self, size):
        self.max_size = size
        self.stack_list = []

    def push(self, value):
        if len(self.stack_list) < self.max_size:
            self.stack_list.append(value)
        else:
            print("Stack is full. Cannot push", value)

    def pop(self):
        if self.stack_list:
            return self.stack_list.pop()
        else:
            print("Stack is empty. Cannot pop.")
            return None

    def peek(self):
        if self.stack_list:
            return self.stack_list[-1]
        else:
            print("Stack is empty. Cannot peek.")
            return None

    def is_empty(self):
        return len(self.stack_list) == 0

    def is_full(self):
        return len(self.stack_list) == self.max_size


stack = Stack(5)

stack.push(10)
stack.push(20)
stack.push(30)

print("Top element:", stack.peek())

print("Popped element:", stack.pop())
print("Popped element:", stack.pop())

print("Is stack empty?", stack.is_empty())
print("Is stack full?", stack.is_full())

File: stack.java

// import java.io.*;

public class stack {
    private int maxSize; // Maximum size of the stack
    private int top; // Index of the top element
    private int[] stackArray; // Array to store the elements

    stack(int size) {
        maxSize = size;
        stackArray = new int[maxSize];
        top = -1; // Initialize top to -1 when the stack is empty
    }

    public void push(int value) {
        if (top < maxSize - 1) {
            stackArray[++top] = value; // Increment top and add the value to the stack
        } else {
            System.out.println("Stack is full. Cannot push " + value);
        }
    }

    public int pop() {
        if (top >= 0) {
            return stackArray[top--]; // Return the top element and decrement top
        } else {
            System.out.println("Stack is empty. Cannot pop.");
            return -1; // Return a default value to indicate stack underflow
        }
    }

    public int peek() {
        if (top >= 0) {
            return stackArray[top]; // Return the top element without removing it
        } else {
            System.out.println("Stack is empty. Cannot peek.");
            return -1; // Return a default value to indicate stack underflow
        }
    }

    public boolean isEmpty() {
        return top == -1; // Stack is empty when top is -1
    }

    public boolean isFull() {
        return top == maxSize - 1; // Stack is full when top is equal to maxSize - 1
    }

    public static void main(String[] args) {
        stack stack = new stack(5);

        stack.push(10);
        stack.push(20);
        stack.push(30);

        System.out.println("Top element: " + stack.peek());

        System.out.println("Popped element: " + stack.pop());
        System.out.println("Popped element: " + stack.pop());

        System.out.println("Is stack empty? " + stack.isEmpty());
        System.out.println("Is stack full? " + stack.isFull());
    }
}
